src/ansys/fluent/core/solver/flobject.py

```diff
@@ -375,13 +375,16 @@ def find_object(self, relative_path):
                 obj = getattr(obj, comp)
         return obj
 
-    def before_execute(self, value):
+    def before_execute(self, value, command_name: Optional[str] = None):
         """Executes before command execution."""
         if hasattr(self, "_do_before_execute"):
-            self._do_before_execute(value)
-            return True
+            base_file_name = self._do_before_execute(value)
+            if command_name in ["read_case_data"]:
+                data_file = value.replace("cas", "dat")
+                self._do_before_execute(data_file)
+            return base_file_name
         else:
-            return False
+            return value
```

Comment:
Last time we were handling this in `upload()` method, now we are handling it in `before_execute()`.

Reply:
I'm thinking about an improvement here. I will get back to you shortly.

Reply:
Done. Thank you. We have tested all scenarios and all of them are working fine. We have also updated relevant test.